docs/source/torch.compiler_faq.md

```diff
@@ -0,0 +1,637 @@
+the communication operations.
+
+## Why is my code crashing?
+
+If your code ran just fine without `torch.compile` and started to
+crash with it is enabled, then the most important first step is figuring
+out which part of the stack your failure occurred. To troubleshoot that,
+follow the steps below and only try the next step if the previous one
+succeeded.
+
+1. `torch.compile(..., backend="eager")` which only runs TorchDynamo
+   forward graph capture and then runs the captured graph with PyTorch.
+   If this fails then there’s an issue with TorchDynamo.
+2. `torch.compile(..., backend="aot_eager")`
+   which runs TorchDynamo to capture a forward graph, and then AOTAutograd
+   to trace the backward graph without any additional backend compiler
+   steps. PyTorch eager will then be used to run the forward and backward
+   graphs. If this fails then there’s an issue with AOTAutograd.
+3. `torch.compile(..., backend="inductor")` which runs TorchDynamo to capture a
+   forward graph, and then AOTAutograd to trace the backward graph with the
+   TorchInductor compiler. If this fails then there’s an issue with TorchInductor
+
+## Why is compilation slow?
+
+- **Dynamo Compilation**– TorchDynamo has a builtin stats function for
+  collecting and displaying the time spent in each compilation phase.
+  These stats can be accessed by calling `torch._dynamo.utils.compile_times()`
+  after executing `torch._dynamo`. By default, this returns a string
+  representation of the compile times spent in each TorchDynamo function by name.
+- **Inductor Compilation**– TorchInductor has a builtin stats and trace function
+  for displaying time spent in each compilation phase, output code, output
+  graph visualization and IR dump. `env TORCH_COMPILE_DEBUG=1 python repro.py`.
+  This is a debugging tool designed to make it easier to debug/understand the
+  internals of TorchInductor with an output that will look something like
+  [this](https://gist.github.com/jansel/f4af078791ad681a0d4094adeb844396)
+  Each file in that debug trace can be enabled/disabled via
+  `torch._inductor.config.trace.*`. The profile and the diagram are both
+  disabled by default since they are expensive to generate. See the
+  [example debug directory
+  output](https://gist.github.com/jansel/f4af078791ad681a0d4094adeb844396)
+  for more examples.
+- **Excessive Recompilation**
+  When TorchDynamo compiles a function (or part of one), it makes certain
+  assumptions about locals and globals in order to allow compiler
+  optimizations, and expresses these assumptions as guards that check
+  particular values at runtime. If any of these guards fail, Dynamo will
+  recompile that function (or part) up to
+  `torch._dynamo.config.recompile_limit` times. If your program is
+  hitting the cache limit, you will first need to determine which guard is
+  failing and what part of your program is triggering it. The
+  [recompilation profiler](#recompilation-profiler) automates the
```

Comment:
#recompilation-profiler brakes the build and it looks like it's [not needed](https://github.com/pytorch/pytorch/issues/141422), so we can say something:

```suggestion
Use `TORCH_TRACE/tlparse` or `TORCH_LOGS=recompiles` to trace the root of the issue.
```
@williamwen42